enterprise/frontend/src/embedding-sdk/components/public/InteractiveQuestion/InteractiveQuestion.tsx

```diff
@@ -56,60 +55,63 @@ export const _InteractiveQuestion = ({
   useEffect(() => {
     const fetchQBData = async () => {
       const { location, params } = getQuestionParameters(questionId);
-      dispatch(initializeQB(location, params));
+      try {
+        await dispatch(initializeQBRaw(location, params));
+      } catch (error) {
+        setLoading(false);
+      }
     };
 
-    fetchQBData().then(() => setLoading(false));
+    fetchQBData();
   }, [dispatch, questionId]);
 
-  if (!loading && !queryResults) {
+  useEffect(() => {
+    if (queryResults) {
+      setLoading(false);
+    }
+  }, [queryResults]);
```

Comment:
Yeah, so `queryStatus` has some quirks that cause things to break still 🙃

Reply:
I see, then the useEffect hook is indeed the best we can do (without modifying their actions/reducers) 😄

Reply:
I tried that one first before using `queryResults` - it seems like `queryStatus` refers more to the actual running of the question, and it comes out as `complete` between the time the component renders and the component starts running the question logic, so there's a tiny bit of time when it would be `complete` before it moves to `running`